File: backend/src/main/java/com/coing/domain/user/controller/dto/UserResponse.java
package com.coing.domain.user.controller.dto;

import java.util.UUID;

import com.coing.domain.user.entity.User;

public record UserResponse(
	UUID id,
	String name,
	String email,
	boolean verified
) {
	public static UserResponse from(User user) {
		return new UserResponse(user.getId(), user.getName(), user.getEmail(), user.isVerified());
	}
}

File: backend/src/main/java/com/coing/domain/user/controller/dto/UserResponse.kt
package com.coing.domain.user.controller.dto

import com.coing.domain.user.entity.Authority
import com.coing.domain.user.entity.User
import com.coing.global.annotation.NoArg
import java.util.UUID

@NoArg
data class UserResponse(
    val id: UUID,
    val name: String,
    val email: String,
    val verified: Boolean,
    val authority: Authority = Authority.ROLE_USER
) {
    companion object {
        fun from(user: User): UserResponse {
            return UserResponse(
                id = user.id!!,
                name = user.name,
                email = user.email,
                verified = user.verified,
                authority = user.authority ?: Authority.ROLE_USER
            )
        }
    }
}
